test/runtests.jl

```diff
@@ -2079,6 +2080,18 @@ Base.OrderStyle(::Type{Num}) = Base.Unordered()
     # Test that @generated functions work with Quantities + custom types (#231)
     @test uconvert(u"°C", Num(373.15)u"K") == Num(100)u"°C"
 end
+area_of_circle(radius::WithDims(u"m")) = pi*radius^2
+area_of_square(side::WithUnits(u"m")) = side^2
+
+@testset "Unitful interfaces" begin
+    @test area_of_circle(Num(1.0)u"m") ≈ pi*m^2
+    @test area_of_circle(Num(1.0)u"km") ≈ pi*km^2
+    @test_throws MethodError area_of_circle(Num(1.0)u"s")
+
+    @test area_of_square(Num(0.5)u"m") ≈ 0.25*m^2
+    @test_throws MethodError area_of_square(Num(0.5)u"km")
+    @test_throws MethodError area_of_square(Num(0.5)u"s")
```

Comment:
These tests can be simplified. The functions `area_of_circle` and `area_of_square` are not necessary, you can just check that the type matches without calling the functions:

```suggestion
@testset "WithDims, WithUnits" begin
    @test isa(Num(1.0)m, WithDims(m))
    @test isa(Num(1.0)km, WithDims(m))
    @test !isa(Num(1.0)s, WithDims(m))

    @test isa(Num(0.5)m, WithUnits(m))
    @test !isa(Num(0.5)km), WithUnits(m))
    @test !isa(Num(0.5)s, WithUnits(m))
```

(I also replaced `u"m"` by `m`. Since the test suite imports these units, the `u"…"` macro is not needed.)

Reply:
Thanks, great idea. Fixed in 15aa3eb8447cba0f8c59401d67bff3efca1d4de3.